### PR TITLE
Fix securing proxy based on published wsdl security policies

### PR DESCRIPTION
**Pull request by himasha**
Introduced a property to specify whether to preserve/remove security policies
of the published wsdl, so that the proxy can specify its own policies.
Fixes wso2/product-ei#1841

## Purpose
When creating a proxy with publishWSDL option, and if the publish wsdl contains security policies, the proxy is automatically secured with those policies. This does not let the user to configure a different security policy to secure the proxy. 

## Goals
Let users add different secured policies to a proxy where the published wsdl already contains security policies of its own.

## Documentation
https://docs.wso2.com/display/EI611/Publishing+a+WSDL  (Need to update with new attribute introduced)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceFactory.java [hunk 1]
```diff
@@ -242,7 +247,8 @@ public static ProxyService createProxy(OMElement elem, Properties properties) {
                         log.error(msg);
                         handleException(msg, e);
                     }
-                } else {
+                }
+                 else {
```
Comment (lahirus) [inline modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceFactory.java]: Minor formatting issue.

Comment (himasha) [reply modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceFactory.java]: Fixed.

File: modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyService.java [hunk 2]
```diff
@@ -448,6 +465,29 @@ public AxisService buildAxisService(SynapseConfiguration synCfg, AxisConfigurati
         // if a WSDL was found
         if (wsdlElement != null) {
             OMNamespace wsdlNamespace = wsdlElement.getNamespace();
+            // if preservePolicy is set to 'false', remove the security policy content of publish wsdl
```
Comment (lahirus) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyService.java]: buildAxisService method is too long. Better to at least take the new part to a different methode.

Comment (himasha) [reply modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyService.java]: +1. Moved code to new method.

**Pull request merged**